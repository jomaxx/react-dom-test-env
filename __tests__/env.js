import React from "react";
import { mount, unmount, waitFor, find, createEnv } from "react-dom-test-env";

afterEach(unmount);

it("create environment", () => {
  createEnv();
});

it("mounts a component", () => {
  mount(<button />);
  expect(find("button")).toBeTruthy();
});

it("unmounts a component", () => {
  mount(<button />);
  unmount();
  expect(find("button")).not.toBeTruthy();
});

it("waits for element", async () => {
  const promise = waitFor("button");
  mount(<button id="test" />);
  const button = await promise;
  expect(button.id).toBe("test");
});

it("throws error after timeout", async () => {
  await expect(waitFor("button", 0)).rejects.toThrowErrorMatchingSnapshot();
});
