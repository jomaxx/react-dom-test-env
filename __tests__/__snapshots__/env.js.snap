// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`throws error after timeout 1`] = `"button not found!"`;
